kubernetes/token.go

```diff
@@ -20,6 +27,32 @@ func GetKialiToken() (string, error) {
 			}
 			KialiToken = string(token)
 		}
+		tokenRead = time.Now()
 	}
 	return KialiToken, nil
 }
+
+// Check if token expired based on the k configuration
+func IsTokenExpired() bool {
+
+	if tokenExpireDuration == 0 {
+		kConfig := kialiConfig.Get()
+		tokenExpireDuration = time.Duration(kConfig.KubernetesConfig.TokenExpireDuration) * time.Second
+	}
+
+	if time.Since(tokenRead) > tokenExpireDuration {
+		return true
+	} else {
+		return false
+	}
+}
+
+// Set ServiceAccountPath
+func SetDefaultServiceAccountPath(path string) {
+	DefaultServiceAccountPath = path
+}
+
+// Set Token Expiration
+func SetTokenExpireDuration(expireTime time.Duration) {
+	tokenExpireDuration = expireTime
+}
```

Comment:
Are these being used anywhere?

Reply:
My eyes see they are being used in the tests.

Reply:
Yes, that is correct

Reply:
Ah well can you set those directly in the tests instead of adding methods to set them since these setters are only used by the tests?
```
DefaultServiceAccountPath = tmpFileTokenExpired
...
```

Reply:
Updated.